src/main/resources/application.yaml

```diff
@@ -14,3 +14,6 @@ idam:
   s2s-auth:
     enabled: ${S2S_ENABLED:true}
     url: ${S2S_URL}
+
+app-insights:
+  dev-mode: ${APPLICATION_INSIGHTS_DEV_MODE:true}
```

Comment:
would be worth to have the key here for consistency and reference here as well (even if Spring doesnt read it)

Reply:
would it not confuse devs? on the other hand - you'll see the env var as a requirement in one place